terraform/lib/dependabot/terraform/file_selector.rb

```diff
@@ -12,6 +12,14 @@ def terragrunt_files
   end
 
   def terragrunt_file?(file_name)
-    file_name != ".terraform.lock.hcl" && file_name.end_with?(".hcl")
+    !lock_file?(file_name) && file_name.end_with?(".hcl")
+  end
+
+  def lock_file?(filename)
+    filename == ".terraform.lock.hcl"
+  end
+
+  def lock_file
+    dependency_files.select { |f| lock_file?(f.name) }
```

Comment:
Could use find to just get the first one matching the method name
```suggestion
    dependency_files.find { |f| lock_file?(f.name) }
```